delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/projects/ProjectsImpl.scala

```diff
@@ -114,11 +117,25 @@ final class ProjectsImpl private (
   private def fetchFromCache(uuid: UUID): IO[ProjectNotFound, ProjectRef] =
     index.collectFirstOr { case (ref, resource) if resource.value.uuid == uuid => ref }(ProjectNotFound(uuid))
 
+  override def provisionProject(subject: Subject): UIO[Unit]              = subject match {
+    case user @ User(subject, realm) if provisioningConfig.enabled =>
+      (for {
+        org       <- IO.fromOption(provisioningConfig.enabledReams.get(realm))
+        proj      <- IO.fromEither(Label.apply(subject)).mapError { err =>
+                       logger.warn(s"Failed to create a project label for $user, due to error '${err.getMessage}'")
+                     }
+        projectRef = ProjectRef(org, proj)
+        exists    <- index.get(projectRef).map(_.isDefined)
+        _         <- provisionIfNotExists(exists, projectRef, user, acls, provisioningConfig)
```

Comment:
Do not pass this exists flag to the `provisionIfNotExists` method, but just do:
```scala
IO.when(!exists)(provisionIfNotExists(projectRef, user, acls, provisioningConfig))
```
and call it `provisionOnNotFound(prooject, user, acls, config)`